tests/integration/security/file_mount_citadel_flow/file_mount_flow_test.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filemountcitadelflow
+
+import (
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+
+	"istio.io/istio/tests/integration/security/util"
+
+	"istio.io/istio/pkg/test/echo/common/scheme"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/tests/integration/security/util/connection"
+)
+
+func TestFileMountCitadelCaFlow(t *testing.T) {
+	framework.NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+
+			istioCfg := istio.DefaultConfigOrFail(t, ctx)
+
+			namespace.ClaimOrFail(t, ctx, istioCfg.SystemNamespace)
+			ns := namespace.NewOrFail(t, ctx, "file-mount-citadel-flow", true)
+
+			var a, b echo.Instance
+			echoboot.NewBuilderOrFail(t, ctx).
+				With(&a, util.EchoConfig("a", ns, false, nil, g, p)).
+				With(&b, util.EchoConfig("b", ns, false, nil, g, p)).
+				BuildOrFail(t)
+
+			checkers := []connection.Checker{
+				{
+					From: a,
+					Options: echo.CallOptions{
+						Target:   b,
+						PortName: "http",
+						Scheme:   scheme.HTTP,
+					},
+					ExpectSuccess: true,
+				},
+			}
+
+			// Sleep 10 seconds for the policy to take effect.
+			time.Sleep(10 * time.Second)
+
+			for _, checker := range checkers {
+				retry.UntilSuccessOrFail(t, checker.Check, retry.Delay(time.Second), retry.Timeout(10*time.Second))
+			}
+
+			// Verify that Citadel Agent is not deployed.
```

Comment:
Wait - why ? You can enable SDS with SPIRE agent or some other SDS agent - Citadel agent is just one implementation of SDS.

Reply:
Currently, when SDS is disabled, Istio is using file mount approach to provision key/cert for workload sidecars. 

Without the change to configmap.yaml, if global.sds.enabled is false but global.sds.udsPath is set, Pilot generate SDS config for Envoy sidecar. Because  Citadel Agent is not running, Envoy cannot get key/cert and workload cannot set up mTLS connection. The test will fail.

Now with this change, this test verifies that the mTLS connection succeeds, and key/cert is not provisioned through Citadel Agent.

The test set up does not enable any SDS agent, so it is guaranteed that file mount approach is taking place. If there are flags that can enable any other SDS agent, please let me know and I will disable them.